src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/BaseConverterOneWay.shared.cs

```diff
@@ -20,14 +25,17 @@ public abstract class BaseConverterOneWay<TFrom, TTo> : ValueConverterExtension,
 		/// <param name="parameter">Additional parameter for the converter to handle. This is not implemented.</param>
 		/// <param name="culture">The culture to use in the converter. This is not implemented.</param>
 		/// <returns>An object of type <see cref="TTo"/>.</returns>
-		public object? Convert(object value, Type targetType, object parameter, CultureInfo culture)
+		public object? Convert(object? value, Type targetType, object parameter, CultureInfo culture)
 		{
-			if (value is not TFrom valueFrom)
-				throw new ArgumentException($"value needs to be of type {typeof(TFrom)}");
-
 			if (targetType != typeof(TTo) && !(typeof(TFrom) != typeof(string)))
 				throw new ArgumentException($"targetType needs to be typeof {typeof(TTo)}");
 
+			if (AllowsNullOrDefault && value is null)
+				return ConvertFrom(default);
```

Comment:
Perhaps if `AllowsNullOrDefault == false && value is null` we could throw an `ArgumentNullException` and mention `AllowsNullOrDefault` in the message to aid developers in understanding how they could fix the issue.

I think this will affect your unit tests but I believe that is a good thing, `ColorNameToColorConverterDisallowingNullConvertInvalidValuesThrowArgumentException` would throw 2 different exceptions based on the `""` and `null` inputs